### PR TITLE
add a functionality to lvol module, to create thin pool / volume on Linux LVM

### PR DESCRIPTION
**Pull request by KlabKatsumi**
On Linux's LVM, thin provisioned volume is supported.

This PR add functionality to create thin pool / volume on Linux's LVM to lvol module.

Linux LVM's thin provisioning volume operation is described here:
https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Logical_Volume_Manager_Administration/thinly_provisioned_volume_creation.html
# 

add `thinpool' parameter.
change`lv' parameter to not required.

To create thin poll, specify `thinpool'(,`vg' and `size') parameter and omit`lv' parameter.
It will create thin pool, nameed by `thinpool' parameter.

To create thin volume, specify both `thinpool' and`lv' parameter (also `vg' and`size' paramter is need).
It will create thin volume on `thinpool' pool, named by`lv' parameter.

Thin volume and pool can delete whith 'state=absent' parameter.
Thin volume is resizable like normal volume.
Thin pool can extend, but not reduce. This limitation is in Linux's LVM.
#

### CODE REVIEW

**Comment by robynbergeron** [comment]
Thanks @KlabKatsumi. @jhoekx @abulimov please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: system/lvol.py [hunk 1]
```diff
@@ -201,8 +213,22 @@ def main():
 
     lvs = parse_lvs(current_lvs)
 
+    if thinpool and lv:
+        for test_lv in lvs:
+            if test_lv['name'] == thinpool:
+                break
+        else:
+            module.fail_json(msg="Thin poll %s does not exist." % thinpool)
```
Comment (mscherer) [inline system/lvol.py:221]: I think the error message is wrong, shouldn't it be "Thin pool" rather than "Thin poll" ?

File: system/lvol.py [hunk 2]
```diff
@@ -123,11 +133,12 @@ def main():
     module = AnsibleModule(
         argument_spec=dict(
             vg=dict(required=True),
-            lv=dict(required=True),
+            lv=dict(type='str'),
```
Comment (mscherer) [inline system/lvol.py:136]: If the argument is no longer required, the doc (line 39) should be changed too.

File: system/lvol.py [hunk 3]
```diff
@@ -201,8 +213,22 @@ def main():
...
+                break
+        else:
+            module.fail_json(msg="Thin poll %s does not exist." % thinpool)
+
+    if thinpool and not lv:
+        target = thinpool
+    elif lv:
+        target = lv
+    else:
+        module.fail_json(msg="lv param is required unless thinpool param is specified.")
```
Comment (mscherer) [inline system/lvol.py:228]: Part of this code can be replaced with "required_one_of" ( see https://github.com/ansible/ansible-modules-core/blob/devel/cloud/digital_ocean/digital_ocean.py#L424 as a example )

File: system/lvol.py (hunk 1 above)
Comment (KlabKatsumi) [inline system/lvol.py:221]: Thank you for reviewing! @mscherer 

Yes. It is miss spelling.

File: system/lvol.py (hunk 2 above)
Comment (KlabKatsumi) [inline system/lvol.py:136]: You are right.

I will correct the doc.

File: system/lvol.py (hunk 3 above)
Comment (KlabKatsumi) [inline system/lvol.py:228]: It's nice!

I will change the code with it in next commit.

**Comment by robynbergeron** [comment]
Thanks @KlabKatsumi for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If we don't hear from you within another 14 days, we will close this pull request.

**Comment by gregdek** [comment]
Thanks @KlabKatsumi for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by smemsh** [comment]
this should be closed as it has been superseded by #2216

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@KlabKatsumi A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Closing due to contributor inactivity.

**Pull request closed without merging**